x-pack/plugins/uptime/public/lib/helper/component_test_helpers.tsx

```diff
@@ -0,0 +1,210 @@
+import { AppState } from '../../state';
+
+/* default mock core */
+const defaultCore = coreMock.createStart();
+const mockCore: () => any = () => {
+  let breadcrumbObj: ChromeBreadcrumb[] = [];
+  const core = {
+    ...defaultCore,
+    application: {
+      getUrlForApp: () => '/app/uptime',
+      navigateToUrl: jest.fn(),
+    },
+    chrome: {
+      setBreadcrumbs: (newBreadcrumbs: ChromeBreadcrumb[]) => {
+        breadcrumbObj = newBreadcrumbs;
+      },
+    },
+  };
+
+  return core;
+};
+
+/* Higher Order Components */
+export function withKibanaContext<T>(
+  WrappedComponent: React.ComponentType<T>,
+  { kibanaProps, customCoreOptions }: { kibanaProps?: { services: any }; customCoreOptions?: any }
+) {
+  const core = {
+    ...mockCore(),
+    customCoreOptions,
+  };
+  return (props: any) => (
+    <KibanaContextProvider services={{ ...core }} {...kibanaProps}>
+      <I18nProvider>
+        <WrappedComponent {...(props as T)} />
+      </I18nProvider>
+    </KibanaContextProvider>
+  );
+}
+
+export function withRouter<T>(WrappedComponent: React.ComponentType<T>, customHistory: History) {
+  const history = customHistory || createMemoryHistory();
+  return (props: T) => (
+    <Router history={history}>
+      <WrappedComponent {...(props as T)} />
+    </Router>
+  );
+}
+
+/* Mock Provider Components */
+export function MockKibanaProvider({
+  children,
+  customCoreOptions,
+  kibanaProps,
+}: {
+  children: ReactElement;
+  customCoreOptions?: any;
+  kibanaProps?: { services: any };
+}) {
+  const core = {
+    ...mockCore(),
+    customCoreOptions,
+  };
+  return (
+    <KibanaContextProvider services={{ ...core }} {...kibanaProps}>
+      <I18nProvider>{children}</I18nProvider>
+    </KibanaContextProvider>
+  );
+}
+
+export function MockRouter({
+  children,
+  customCoreOptions,
+  customHistory,
+  kibanaProps,
+}: {
+  children: ReactElement;
+  customCoreOptions?: any;
+  customHistory?: History;
+  kibanaProps?: { services: any };
+}) {
+  const history = customHistory || createMemoryHistory();
+  return (
+    <Router history={history}>
+      <MockKibanaProvider customCoreOptions={customCoreOptions} kibanaProps={kibanaProps}>
+        {children}
+      </MockKibanaProvider>
+    </Router>
+  );
+}
+
+/* React testing library custom render functions */
+export const renderTLWithKibana = (
+  ui: ReactElement,
+  {
+    customCoreOptions,
+    kibanaProps,
+    renderOptions,
+  }: {
+    customCoreOptions?: any;
+    kibanaProps?: { services: any };
+    renderOptions?: any;
+  } = {}
+) => {
+  return render(
+    <MockKibanaProvider {...kibanaProps} customCoreOptions={customCoreOptions}>
+      {ui}
+    </MockKibanaProvider>,
+    renderOptions
+  );
+};
+
+export const renderTLWithRouter = (
+  ui: ReactElement,
+  {
+    customHistory,
+    customCoreOptions,
+    kibanaProps,
+    renderOptions,
+  }: {
+    customHistory?: History;
+    customCoreOptions?: any;
+    kibanaProps?: { services: any };
```

Comment:
I'm not sure what the types should be for some of these pieces. I welcome feedback here.

Reply:
let's remove the custom prefix, i think that's unnecessary

Reply:
Regarding the types, I believe Kibana Core team exports types for core that we can use here. See https://github.com/elastic/kibana/blob/master/x-pack/plugins/uptime/public/apps/plugin.ts#L7

cc @shahzad31 does that sound right?

Reply:
i think there might be a kibana core mock as well exported by team, may be use that directly?

Reply:
i have seen that somewhere

Reply:
@dominiqueclarke might be worth asking in #kibana-developer-experience channel